src/main/java/gregtech/api/fluids/MetaFluids.java

```diff
@@ -0,0 +1,253 @@
+        setMaterialFluidTexture(Materials.Naphtha, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.LightFuel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.HeavyFuel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.LPG, FluidTypes.GAS);
+        setMaterialFluidTexture(Materials.LightlySteamCrackedLightFuel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.SeverelySteamCrackedLightFuel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.LightlySteamCrackedHeavyFuel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.SeverelySteamCrackedHeavyFuel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Chlorine, FluidTypes.GAS);
+        setMaterialFluidTexture(Materials.CetaneBoostedDiesel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.SodiumPersulfate, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.GlycerylTrinitrate, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Lubricant, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Creosote, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.SeedOil, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Oil, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Diesel, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Biomass, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Ethanol, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.SulfuricAcid, FluidTypes.ACID);
+        setMaterialFluidTexture(Materials.Milk, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.McGuffium239, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Glue, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.HydrochloricAcid, FluidTypes.ACID);
+        setMaterialFluidTexture(Materials.LeadZincSolution, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.NaturalGas, FluidTypes.GAS);
+        setMaterialFluidTexture(Materials.Blaze, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.FluoroantimonicAcid, FluidTypes.ACID);
+        setMaterialFluidTexture(Materials.Naquadah, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.NaquadahEnriched, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Naquadria, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.Ice, FluidTypes.LIQUID);
+        setMaterialFluidTexture(Materials.UUMatter, FluidTypes.LIQUID);
+    }
+
+    /**
+     * Every {@link gregtech.api.unification.material.info.MaterialIconSet} that has fluids requires a registered sprite
+     */
+    private static void registerIconFluidSprites() {
+        for (MaterialIconSet materialIconSet : MaterialIconSet.ICON_SETS.values()) {
+            fluidSprites.add(new ResourceLocation(GTValues.MODID, "blocks/material_sets/" + materialIconSet.name.toLowerCase(Locale.ROOT) + "/fluid"));
```

Comment:
No need to do `toLowerCase` here since all `MaterialIconSet` values are now canonically lowercase.